crud/select/compat/common.lua

```diff
@@ -1,6 +1,20 @@
+local ratelimit = require('crud.ratelimit')
+local check_select_args_rl = ratelimit:new()
+
 local common = {}
 
 common.SELECT_FUNC_NAME = '_crud.select_on_storage'
 common.DEFAULT_BATCH_SIZE = 100
 
+common.check_select_args = function(space_name, user_conditions, opts)
+    local rl = check_select_args_rl
+    local uc = user_conditions
+
+    if (uc == nil or (type(uc) == 'table' and next(uc) == nil)) and
+       (opts == nil or not opts.fullscan) then
+        rl:log_crit("empty or nil `select` call on user space=%s\n %s",
+                    space_name, debug.traceback())
+    end
```

Comment:
Yet another design flaw: we can set a condition by a non-indexed field (or a field that is not the first one in any index) and get full scan without the warning. Asked products managers...

Reply:
requirements have been updated:
https://github.com/tarantool/tarantool/issues/7129

Reply:
It is still relevant. Tarantool itself has pretty simple query language, but crud's one a bit more powerful. It requires more analysis to actually say, whether the query will fall into the linear space scan. I propose to consider using the query plan information.

Reply:
Updated.